### PR TITLE
[Quick PR] Add authz authorizations for core modules param changes

### PR DESCRIPTION
**Pull request by okdas**
Adds param changes authorizations for DAO.

## Summary by CodeRabbit

- **New Features**
	- Enhanced authorization framework with multiple new entries, allowing for granular permission settings across various functionalities in the Cosmos ecosystem, including software upgrades and governance management.
	- Extended authorization capabilities for managing interactions within the Cosmos blockchain.

These improvements aim to provide users with a more flexible and robust permission management system.

### CODE REVIEW

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 1]
```diff
@@ -97,5 +97,104 @@
       "msg": "\/poktroll.shared.MsgUpdateParam"
     },
     "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.upgrade.v1beta1.MsgSoftwareUpgrade"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:106]: Plans an upgrade (implemented in #702)

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 2]
```diff
@@ -97,5 +97,104 @@
...
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.upgrade.v1beta1.MsgCancelUpgrade"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:115]: Cancels an existing upgrade (#702)

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 3]
```diff
@@ -97,5 +97,104 @@
...
+      "msg": "\/cosmos.mint.v1beta1.MsgUpdateParams"
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.staking.v1beta1.MsgUpdateParams"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:160]: Exposed params:

```yaml
params:
  bond_denom: upokt
  historical_entries: 10000
  max_entries: 7
  max_validators: 1
  min_commission_rate: "0"
  unbonding_time: 504h0m0s
```

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 4]
```diff
@@ -97,5 +97,104 @@
...
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.slashing.v1beta1.MsgUpdateParams"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:169]: Exposed params:

```yaml
params:
  downtime_jail_duration: 10m0s
  min_signed_per_window: "0.500000000000000000"
  signed_blocks_window: "100"
  slash_fraction_double_sign: "0.050000000000000000"
  slash_fraction_downtime: "0.010000000000000000"
```

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 5]
```diff
@@ -97,5 +97,104 @@
...
+      "msg": "\/cosmos.distribution.v1beta1.MsgUpdateParams"
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.mint.v1beta1.MsgUpdateParams"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:151]: Exposed params:
```yaml
params:
  blocks_per_year: "6311520"
  goal_bonded: "670000000000000000"
  inflation_max: "200000000000000000"
  inflation_min: "70000000000000000"
  inflation_rate_change: "130000000000000000"
  mint_denom: upokt
```

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 6]
```diff
@@ -97,5 +97,104 @@
...
+      "msg": "\/cosmos.upgrade.v1beta1.MsgCancelUpgrade"
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.auth.v1beta1.MsgUpdateParams"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:124]: Exposed params:
```yaml
params:
  max_memo_characters: "256"
  sig_verify_cost_ed25519: "590"
  sig_verify_cost_secp256k1: "1000"
  tx_sig_limit: "7"
  tx_size_cost_per_byte: "10"
```

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 7]
```diff
@@ -97,5 +97,104 @@
...
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.bank.v1beta1.MsgUpdateParams"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:133]: Exposed params:
```yaml
params:
  default_send_enabled: true
```
(can whitelist denoms on network that way)

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 8]
```diff
@@ -97,5 +97,104 @@
...
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.distribution.v1beta1.MsgUpdateParams"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:142]: Exposed params: https://github.com/cosmos/cosmos-sdk/blob/main/x/distribution/proto/cosmos/distribution/v1beta1/distribution.proto#L13

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 9]
```diff
@@ -97,5 +97,104 @@
...
+      "msg": "\/cosmos.slashing.v1beta1.MsgUpdateParams"
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.gov.v1.MsgUpdateParams"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:178]: Exposed params: https://github.com/cosmos/cosmos-sdk/blob/main/x/gov/proto/cosmos/gov/v1/gov.proto#L260

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 10]
```diff
@@ -97,5 +97,104 @@
...
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.gov.v1.MsgCancelProposal"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:187]: Allows to cancel an existing proposal. I don't know what the plan is with the `gov` module yet - as it is still available and anybody can submit a proposal. Just want us to also give an option to cancel it.

File: tools/scripts/authz/dao_genesis_authorizations.json [hunk 11]
```diff
@@ -97,5 +97,104 @@
...
+    },
+    "expiration": "2500-01-01T00:00:00Z"
+  },
+  {
+    "granter": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+    "grantee": "pokt1eeeksh2tvkh7wzmfrljnhw4wrhs55lcuvmekkw",
+    "authorization": {
+      "@type": "\/cosmos.authz.v1beta1.GenericAuthorization",
+      "msg": "\/cosmos.consensus.v1.MsgUpdateParams"
```
Comment (okdas) [inline tools/scripts/authz/dao_genesis_authorizations.json:196]: Exposed CometBFT params: https://github.com/cosmos/cosmos-sdk/blob/main/x/consensus/proto/cosmos/consensus/v1/tx.proto#L28

**Review by Olshansk — APPROVED** [review]

**Comment by okdas** [comment]
Merging with a couple of flaky tests failing.

**Pull request merged**